sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
 
 from azure.core.pipeline.policies import HttpLoggingPolicy
 
+from .http_constants import HttpHeaders
 
 
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs) -> None: # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
         if self._enable_diagnostics_logging:
-            self.logger = logger
-
-    def on_request(self, request): # pylint: disable=too-many-return-statements, too-many-statements
-        # type: (PipelineRequest) -> None
-        http_request = request.http_request
-        options = request.context.options
-        self._enable_diagnostics_logging = request.context.setdefault(
-            "enable_diagnostics_logging",
-            options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging))
-        if self._enable_diagnostics_logging:
-            # Get logger in my context first (request has been retried)
-            # then read from kwargs (pop if that's the case)
-            # then use my instance logger
-            logger = request.context.setdefault("logger", options.pop("logger", self.logger))
-
+            cosmos_disallow_list = ["Authorization"]
```

Comment:
```suggestion
            cosmos_disallow_list = ["Authorization", "ProxyAuthorization"]
```
there is one other authorization header, suggesting it here since they're the only 2 we should be blocking